Add the passing of commit hash to the config file.

CameronDiver opened the pull request:
The hash of the commit is retrieved by git, and parsed from the yaml, then passed through as normal to the configuration file.

Change-Type: minor
Connects-To: #50

CameronDiver commented:
Tests also need to be changed as a lot of them don't expect to be also retrieving a hash.

jviotti commented on lib/git-log.js:
```diff
@@ -38,7 +38,9 @@ const tags = require('./tags');
  */
 exports.GIT_LOG_YAML_PRETTY_FORMAT = [
-  '- subject: >-',
+  '- hash: >-',
```
The hash is a single string, so we shouldn't be using YAML multi-string support here (the `>-`). Something like `- hash: %H` should suffice.

CameronDiver replied:
Ah that makes sense :) I've not really used yaml much.

jviotti commented:
> Tests also need to be changed as a lot of them don't expect to be also retrieving a hash.

Yep. Lets also add some tests for this functionality as well.

jviotti commented on tests/git-log.spec.js:
```diff
@@ -183,6 +183,7 @@ describe('GitLog', function() {
             'Currently we had AppImage scripts and other resources in various',
             'different places in the code base.'
           ].join('\n'),
+          hash: undefined,
```
Even better: why not updating the `utils.formatCommit` function to also accept a hash, and check that is propagated correctly?

jviotti commented on lib/git-log.js:
```diff
@@ -38,7 +38,9 @@ const tags = require('./tags');
 exports.GIT_LOG_YAML_PRETTY_FORMAT = [
-  '- subject: >-',
+  '- hash:',
+  '    %H',
```
Is there any reason why `%H` is in another line? Now that we're not using the multi line modified, this can be simply `- hash: %H`.

jviotti commented:
I left a minor comment in the YAML string, but will approve already :+1:

lurch commented on tests/utils.js:
```diff
@@ -26,19 +26,22 @@ const indentString = require('indent-string');
  *
  * @example
  * const formattedCommit = utils.formatCommit({
+ *   hash: b8ca72ee58bddc9d174b71615aaa0159849b2aca
```
Example hash here is in lowercase, but all the examples have their hashes in uppercase?

lurch commented:
Maybe a silly question, but is it worth adding a unit-test that the hash retrieved from git is in the correct format? (only hex-chars, specific length, all lowercase, etc.)

This PR doesn't include any modifications to the documentation - should it?

Only a nitpick, but I guess "then passed through as normal to the configuration file." in the commit-message actually means "then made available as an attribute on the `commit` objects in the configuration file." ?

CameronDiver commented:
I've added documentation and a single object in the parse function which is updated and returned as necessary. Also added into the readme information about the new commit object which is provided to the conf file. Testing the output of the git hash doesn't really make sense to me as it doesn't matter what git gives us back as long as it can address the commits. I agree with you on the commit message, I'll do a rebase before merge and make it more appropriate.

lurch commented:
On a side note, I guess it might be useful if the [documentation](https://github.com/resin-io/versionist/blob/master/README.md#template-string) explicitly stated what attributes are available in the `commit` objects given to the template, rather than simply 'inferring' it from the [parseFooterTags](https://github.com/resin-io/versionist/blob/master/README.md#parsefootertags-boolean) docs.
